src/MatchedLearnerApi.Application.Tests/MappersTests/MatchedLearnerDtoMapperTests/MapTestsWithMultipleDataLockEventsForOneApprenticeship.cs

```diff
@@ -0,0 +1,402 @@
+        {
+            var sut = new MatchedLearnerDtoMapper();
+
+            var actual = sut.Map(_testInput);
+            actual.Training.First().PriceEpisodes.Should().HaveCount(2);
+            actual.Training.First().PriceEpisodes.Should().ContainEquivalentOf(new { Identifier = "1-1-1" });
+            actual.Training.First().PriceEpisodes.Should().ContainEquivalentOf(new { Identifier = "2-2-2" });
+            actual.Training.Last().PriceEpisodes.Should().HaveCount(2);
+            actual.Training.Last().PriceEpisodes.Should().ContainEquivalentOf(new { Identifier = "3-3-3" });
+            actual.Training.Last().PriceEpisodes.Should().ContainEquivalentOf(new { Identifier = "4-4-4" });
+        }
+
+        [Test]
+        public void Training_Should_BeInOutput()
+        {
+            var sut = new MatchedLearnerDtoMapper();
+
+            var actual = sut.Map(_testInput);
+
+            actual.Training.Should().ContainEquivalentOf(new { PathwayCode = 1 });
+        }
+
+        [Test]
+        public void ThereShouldBe_NoMixingOfPeriodsBetweenPriceEpisodes()
+        {
+            var sut = new MatchedLearnerDtoMapper();
+
+            var actual = sut.Map(_testInput);
+
+            var firstEvent = actual.Training.FirstOrDefault(x => x.PathwayCode == 1);
+            firstEvent!.PriceEpisodes.SelectMany(x => x.Periods).Should().HaveCount(2);
+            firstEvent.PriceEpisodes.SelectMany(x => x.Periods).Should().ContainEquivalentOf(new { Period = 2 });
+            firstEvent.PriceEpisodes.SelectMany(x => x.Periods).Should().ContainEquivalentOf(new { Period = 1 });
+        }
+
+        //TODO: Move to Repository Tests
+        [Test]
+        public void EventsFromSameApprenticeships_Should_BeGrouped()
+        {
+            var testInput = new List<DatalockEvent>
+            {
+                new DatalockEvent
+                {
+                    AcademicYear = 2021,
+                    LearningAimPathwayCode = 1,
+                    LearningAimStandardCode = 2,
+                    LearningAimFrameworkCode = 3,
+                    LearningAimProgrammeType = 4,
+                    LearningAimReference = "123",
+                    PriceEpisodes = new List<DatalockEventPriceEpisode>(),
+                },
+                new DatalockEvent
+                {
+                    AcademicYear = 2021,
+                    LearningAimPathwayCode = 1,
+                    LearningAimStandardCode = 2,
+                    LearningAimFrameworkCode = 3,
+                    LearningAimProgrammeType = 4,
+                    LearningAimReference = "123",
+                    PriceEpisodes = new List<DatalockEventPriceEpisode>(),
+                }
+            };
+
+            var sut = new MatchedLearnerDtoMapper();
+
+            var actual = sut.Map(testInput);
+
+            actual.Training.Should().HaveCount(2);
+        }
+
+        //TODO: Move to Repository Tests
```

Comment:
Should this be moved? Looking at the test it is creating a list of two items and passing them to a mapper which does not grouping, then asserting that the count of the result is 2.

Reply:
I am working on this in another branch